### DISCUSSION TITLE
How to get BitsPerPixel of any image by using imagemagick?

### DISCUSSION THREAD

**Discussion by pme9** [ImageMagick/ImageMagick#5005] [Help]
I'm reading one image and I want to get bits per pixel of that image.
How to get BitsPerPixel of any image by using imagemagick?

Thanks for the help!

**Comment by snibgo** [comment]
I don't understand the question.

DLLs (Dynamic Link Libraries) can't be used directly.

ImageMagick is used by issuing commands, or writing scripts or programs.

**Reply by pme9** [reply]
sorry, My bad!
Actually I'm reading one image using ImageMagick. But I want to get bits per pixel data of that image. 
Can you please tell me how to do that?

**Reply by dlemstra** [reply]
How are you reading the image? Are you using the command line or some kind of library?

**Reply by pme9** [reply]
By using read function of imagemagick like below:

Magick::Image image;
image.read(filePath);

**Comment by snibgo** [comment]
If you want bits per channel per pixel, ImageMagick calls that property "depth". So, for example:
```
    cout << image.depth();
```
If you want the total number of bits per pixel, multiply that by the number of channels.
